doc/source/whatsnew/v0.25.0.rst

```diff
@@ -425,6 +425,7 @@ Indexing
 - Bug in which :meth:`DataFrame.to_csv` caused a segfault for a reindexed data frame, when the indices were single-level :class:`MultiIndex` (:issue:`26303`).
 - Fixed bug where assigning a :class:`arrays.PandasArray` to a :class:`pandas.core.frame.DataFrame` would raise error (:issue:`26390`)
 - Allow keyword arguments for callable local reference used in the :method:`DataFrame.query` string (:issue:`26426`)
+- Bug in assignment to multiple columns of a `DataFrame` when some of the columns do not exist (:issue:`13658`)
```

Comment:
use ``:class:`DataFrame` ``; this is assignment with a scalar, yes?

Reply:
The assignment should allow the same set of values as when the columns are present.

Reply:
Hi, thanks for the feedback. Any pointer on how to streamline the code?

Reply:
> use `` :class:`DataFrame` ``

can you update.

Reply:
Updated